dist/s_whitespace

```diff
@@ -1,18 +1,24 @@
 #! /bin/sh
 
-# Single space and remove trailing whitespace from source files.
+# Single space, newline at the end of file, and remove trailing whitespace from source files.
 t=__wt.$$
 trap 'rm -f $t' 0 1 2 3 13 15
 
 # Clear lines that only contain whitespace, compress multiple empty lines
 # into a single line, discard trailing empty lines.
+# Add a newline to the end of the file if there is not one.
 whitespace()
 {
 	! head $1 | grep -q 'automatically generated by SWIG' || return
 	sed -e 's/[	 ][	 ]*$//' < $1 | \
 	    cat -s | \
 	    sed -e '${' -e '/^$/d' -e '}' > $t
 	cmp $t $1 > /dev/null 2>&1 || (echo "$1" && cp $t $1)
+    if (test "$(tail -c 1 "$1" | wc -l)" -eq 0)
+    then
+        echo "Adding newline at end of file: $f"
+    fi
+    sed -i -e '$a\' $1
```

Comment:
So we just run this for everything and presumably it only modifies the things it needs to? Does its return code indicate whether it did anything?

Reply:
Yes this runs on everything but it only adds a newline to the end of a file if there isn't one. So if a file has a newline nothing will be added. There is no return code but I used the above check to see if a file does not have a newline; so any file that has been modified will be printed.

Reply:
Thanks @keithbostic ! I have updated the function with those changes.